src/backends/gpu/kernels/reduce_by_key.hpp

```diff
@@ -22,6 +59,9 @@ unsigned rounddown_power_of_2(std::uint32_t i) {
 // with two 32 bit shuffles.
 // get_from_lane is a wrapper around __shfl() for both single and double
 // precision.
+// TODO: CUDA 9 provides a double precision version of __shfl that
```

Comment:
Why not know with an #if test? Since anyhow we have a bunch of preprocessor tests for this?

Reply:
Because it is another `#if` test, and I would rather just replace this wholesale when the time comes to use CUDA9.